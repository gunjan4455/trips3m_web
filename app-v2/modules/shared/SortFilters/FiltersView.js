import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import styled from '@emotion/styled';

import { DownArrow, CloseDark, FilterIconWhite } from 'helpers/Icon/Icon';

const GFilterButton = styled.button({
  width: '56px',
  height: '56px',
  boxShadow: '0 4px 12px #999',
  '& svg': {
    position: 'absolute',
    width: '25px',
    height: '25px',
  }
});

const GButton = styled.button({
  width: '24px',
  height: '24px',
  '& svg': {
    width: '16px',
    height: '16px',
  }
});

const FilterComponent = ({ name, type, className, id, disabled, input, checked }) => {
  return (
    <input {...input} name={name} type={type} disabled={disabled} id={id} className={className} />
  );
};

FilterComponent.propTypes = {
  name: PropTypes.string,
  type: PropTypes.string,
  className: PropTypes.string,
  id: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  input: PropTypes.object.isRequired,
  checked: PropTypes.bool.isRequired
};

export const FiltersView = ({ handleSubmit, filterList, filterToggle, initialValues, resetFilters }) => (
  <div>
    <GFilterButton className="overflowh hide pbc1 radius100 fixed z14 r15 b15 flex alignCenter justifyCenter">
      <FilterIconWhite />
    </GFilterButton>
    <div className='col-xs-12 p0 z15 fixed t0 l0 overflowa pb64 b0 r0 sbc5 cityblockfilterContainer'>
      <div className="row row- sbcw filters-options-con">
        <div className="row row- sbcw p15 flex alignCenter filters-header">
          <div className="col-xs-2 pl0">
            <GButton className="close-filters flex alignCenter justifyCenter" type="button" onClick={filterToggle}><CloseDark /></GButton>
          </div>
          <div className="col-xs-6 pl8">
            <p className="filter-title">Filters</p>
          </div>
          <div className="col-xs-4 pl0 pr0">
            <button className="reset-button fright" onClick={resetFilters}>RESET</button>
          </div>
        </div>

        <div className="pt8 sbc5" />
        <form className="filter-row-container block">
          {filterList && filterList.length && filterList
          .filter(filter => filter.options.length)
          .map((filter, index) => (
            <div className="at_filtersrow p15 pb0 sbcw filter-row" key={index}>
              <h3 className="destination-type-btn fw7 mb15 relative">
                {filter.title}
                <span className="block absolute-center-v"><DownArrow className="smooth" /></span>
              </h3>
              <div className="row flex spaceBetween flexWrap destination-type-body">
                {filter.options.map((filterOption, key) => (
                  <div className="col-xs-6" key={key}>
                    <Field
                      name={`${filter.paramName}[${filterOption.value}]`}
                      component={FilterComponent}
                      id={`${filter.paramName}[${filterOption.value}]`}
                      type="checkbox"
                      className="checkbox-common"
                      disabled={!filterOption.enabled}
                      checked={filterOption.selected}
                    />
                    <label htmlFor={`${filter.paramName}[${filterOption.value}]`}>{filterOption.title}</label>
                  </div>))
                }
              </div>
              <div className="row pt8 mt15 sbc5" />
            </div>))
          }
          <div className="fixed p8 sbcw b0 l0 r0 bs4">
            <button className="btn-filled-pri-large wfull" type="button" onClick={handleSubmit}>Apply Filter</button>
          </div>
        </form>
      </div>
    </div>
  </div>
);

FiltersView.propTypes = {
  initialValues: PropTypes.object.isRequired,
  handleSubmit: PropTypes.func,
  resetFilters: PropTypes.func,
  filterToggle: PropTypes.func,
  filterList: PropTypes.array
};


FiltersView.defaultProp = {
};

export default reduxForm({
  form: 'hlpFilters',
  initialValues: { enabled: true }
})(FiltersView);
