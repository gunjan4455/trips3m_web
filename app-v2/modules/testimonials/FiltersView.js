import React from 'react';
import PropTypes from 'prop-types';

import RatingStar from 'modules/shared/RatingStar';
import { CloseDark } from 'helpers/Icon/Icon';

class FiltersView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedFilters: props.initialFilters,
      ratingFilters: props.ratingFilters,
    };
  }

  checkStatus = (e, option, i) => {
    const ratingFilter = this.state.ratingFilters.map((ratingFilter, index) => {
      if (index !== i) {
        return ratingFilter;
      } else {
        return { ...ratingFilter, checked: e.target.checked };
      }
    });

    let filter = [...this.state.selectedFilters];
    if (e.target.checked) {
      filter.push(option);
    } else {
      filter = filter.filter(
        rating => rating !== option
      );
    }
    this.setState({
      selectedFilters: filter,
      ratingFilters: ratingFilter,
    });
  };

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.setState({
      selectedFilters: nextProps.initialFilters,
      ratingFilters: nextProps.ratingFilters,
    });
  };

  handleSubmit = () => {
    const selectedFilters = [...this.state.selectedFilters];
    let testimonialUrl = this.props.location.pathname;
    let filterUrl = '';
    filterUrl += selectedFilters
      .map((filter, i) => {
        if (i > 0) {
          return (`&ratings[]=${filter}`);
        }
      })
      .join('');
    filterUrl = selectedFilters.length ? `ratings[]=${selectedFilters[0]}` + filterUrl : '';
    filterUrl ? testimonialUrl += `?${filterUrl}` : testimonialUrl;
    this.props.updateUrl(testimonialUrl);
  };

  resetFilters = () => {
    const ratingFilter = this.state.ratingFilters.map(ratingFilter => ({ ...ratingFilter }));
    ratingFilter.map(rating => {
      return (rating.checked = false);
    });
    this.setState({
      ratingFilters: ratingFilter,
      selectedFilters: [],
    });
  };

  render() {

    const { filterToggle } = this.props;

    return (
      <div className="fixed t0 l0 sbcw z16 wfull pb48 hfull overflowa">
        <div className="p15 bb">
          <div className='at_close iblock icon-14' onClick={filterToggle}>
            <CloseDark />
          </div>

          <div className="f16 fw4 pfc3 pl24 iblock">Filters</div>
          <div className='at_reset fright f14 pfc4 text-uppercase fw4' onClick={this.resetFilters}>Reset</div>
        </div>
        <div className="p15">
          <div className="fw7 pfc3 f14 pb15">Traveler Rating</div>
          <div className="row">
            {
              this.state.ratingFilters.map((data, i) => {
                return (
                  <div key={data.value} className="col-xs-6">
                    <input
                      type="checkbox"
                      checked={data.checked}
                      name={`[${data.value}]`}
                      id={`[${data.value}]`}
                      onChange={e => this.checkStatus(e, data.value, i)}
                      className="checkbox-common"
                    />
                    <label htmlFor={`[${data.value}]`}>
                      <RatingStar rating={parseInt(data.value)} />
                    </label>
                  </div>
                );
              })
            }
          </div>
        </div>
        <div className="fixed p8 wfull b0 bs6 sbcw z1">
          <button type="button" className='at_handleSubmit wfull f14 fw7 sfcw pbc2 input-with-icon-box'
                  onClick={this.handleSubmit}>
            Apply Filter
          </button>
        </div>
      </div>
    );
  }
}

FiltersView.propTypes = {
  ratingFilters: PropTypes.array.isRequired,
  location: PropTypes.object.isRequired,
  updateUrl: PropTypes.func.isRequired,
  initialFilters: PropTypes.array,
  filterToggle: PropTypes.func.isRequired,
};

FiltersView.defaultProps = {
  initialFilters: []
};

export default FiltersView;
