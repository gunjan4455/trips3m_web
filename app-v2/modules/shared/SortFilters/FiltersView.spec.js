import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import { FiltersView } from './FiltersView';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  initialValues:{},
  filterList: []
};

const shallowWrappperFiltersView = (props) => {
  return shallow(<FiltersView {...props} />);
};

describe('Pdm: Render Correctly ', () => {

  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should match with the previous snapshot', () => {
    const wrapper = shallowWrappperFiltersView(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render filter list when filterlist is not empty', () => {
    const wrapper = shallowWrappperFiltersView(props);
    expect(wrapper.find('.at_filtersrow')).toHaveLength(0);
  });

  it('should render only one filter from list when options in empty', () => {
    props.filterList.push({
      id:8137,
      multiSelect: true,
      operation: 'OR',
      options: [{
        count:177,
        enabled: true,
        id:8138,
        selected: false,
        title:'5 Star',
        value: '5'
      }],
      paramName: 'star_ratings',
      title: 'Star Rating'
    })
    const wrapper = shallowWrappperFiltersView(props);
    expect(wrapper.find('.at_filtersrow')).toHaveLength(1);
  });

  it('should render filter list when filterlist is not empty', () => {
    props.filterList.push({
      id: 8138,
      multiSelect: true,
      operation: 'OR',
      options: [],
      paramName: 'star_ratings',
      title: 'Star Rating'
    })
    const wrapper = shallowWrappperFiltersView(props);
    expect(wrapper.find('.at_filtersrow')).toHaveLength(1);
  });

});
