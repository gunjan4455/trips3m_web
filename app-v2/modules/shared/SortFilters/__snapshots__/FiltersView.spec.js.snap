// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pdm: Render Correctly  should match with the previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FiltersView
    filterList={Array []}
    initialValues={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef(render)
          className="overflowh hide pbc1 radius100 fixed z14 r15 b15 flex alignCenter justifyCenter"
        >
          <IconSvg />
        </ForwardRef(render)>,
        <div
          className="col-xs-12 p0 z15 fixed t0 l0 overflowa pb64 b0 r0 sbc5 cityblockfilterContainer"
        >
          <div
            className="row row- sbcw filters-options-con"
          >
            <div
              className="row row- sbcw p15 flex alignCenter filters-header"
            >
              <div
                className="col-xs-2 pl0"
              >
                <ForwardRef(render)
                  className="close-filters flex alignCenter justifyCenter"
                  type="button"
                >
                  <IconSvg />
                </ForwardRef(render)>
              </div>
              <div
                className="col-xs-6 pl8"
              >
                <p
                  className="filter-title"
                >
                  Filters
                </p>
              </div>
              <div
                className="col-xs-4 pl0 pr0"
              >
                <button
                  className="reset-button fright"
                >
                  RESET
                </button>
              </div>
            </div>
            <div
              className="pt8 sbc5"
            />
            <form
              className="filter-row-container block"
            >
              0
              <div
                className="fixed p8 sbcw b0 l0 r0 bs4"
              >
                <button
                  className="btn-filled-pri-large wfull"
                  type="button"
                >
                  Apply Filter
                </button>
              </div>
            </form>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <IconSvg />,
          "className": "overflowh hide pbc1 radius100 fixed z14 r15 b15 flex alignCenter justifyCenter",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "button",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            Object {
              "& svg": Object {
                "height": "25px",
                "position": "absolute",
                "width": "25px",
              },
              "boxShadow": "0 4px 12px #999",
              "height": "56px",
              "width": "56px",
            },
          ],
          "defaultProps": undefined,
          "displayName": "Styled(button)",
          "render": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row row- sbcw filters-options-con"
          >
            <div
              className="row row- sbcw p15 flex alignCenter filters-header"
            >
              <div
                className="col-xs-2 pl0"
              >
                <ForwardRef(render)
                  className="close-filters flex alignCenter justifyCenter"
                  type="button"
                >
                  <IconSvg />
                </ForwardRef(render)>
              </div>
              <div
                className="col-xs-6 pl8"
              >
                <p
                  className="filter-title"
                >
                  Filters
                </p>
              </div>
              <div
                className="col-xs-4 pl0 pr0"
              >
                <button
                  className="reset-button fright"
                >
                  RESET
                </button>
              </div>
            </div>
            <div
              className="pt8 sbc5"
            />
            <form
              className="filter-row-container block"
            >
              0
              <div
                className="fixed p8 sbcw b0 l0 r0 bs4"
              >
                <button
                  className="btn-filled-pri-large wfull"
                  type="button"
                >
                  Apply Filter
                </button>
              </div>
            </form>
          </div>,
          "className": "col-xs-12 p0 z15 fixed t0 l0 overflowa pb64 b0 r0 sbc5 cityblockfilterContainer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="row row- sbcw p15 flex alignCenter filters-header"
              >
                <div
                  className="col-xs-2 pl0"
                >
                  <ForwardRef(render)
                    className="close-filters flex alignCenter justifyCenter"
                    type="button"
                  >
                    <IconSvg />
                  </ForwardRef(render)>
                </div>
                <div
                  className="col-xs-6 pl8"
                >
                  <p
                    className="filter-title"
                  >
                    Filters
                  </p>
                </div>
                <div
                  className="col-xs-4 pl0 pr0"
                >
                  <button
                    className="reset-button fright"
                  >
                    RESET
                  </button>
                </div>
              </div>,
              <div
                className="pt8 sbc5"
              />,
              <form
                className="filter-row-container block"
              >
                0
                <div
                  className="fixed p8 sbcw b0 l0 r0 bs4"
                >
                  <button
                    className="btn-filled-pri-large wfull"
                    type="button"
                  >
                    Apply Filter
                  </button>
                </div>
              </form>,
            ],
            "className": "row row- sbcw filters-options-con",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="col-xs-2 pl0"
                  >
                    <ForwardRef(render)
                      className="close-filters flex alignCenter justifyCenter"
                      type="button"
                    >
                      <IconSvg />
                    </ForwardRef(render)>
                  </div>,
                  <div
                    className="col-xs-6 pl8"
                  >
                    <p
                      className="filter-title"
                    >
                      Filters
                    </p>
                  </div>,
                  <div
                    className="col-xs-4 pl0 pr0"
                  >
                    <button
                      className="reset-button fright"
                    >
                      RESET
                    </button>
                  </div>,
                ],
                "className": "row row- sbcw p15 flex alignCenter filters-header",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <ForwardRef(render)
                      className="close-filters flex alignCenter justifyCenter"
                      type="button"
                    >
                      <IconSvg />
                    </ForwardRef(render)>,
                    "className": "col-xs-2 pl0",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <IconSvg />,
                      "className": "close-filters flex alignCenter justifyCenter",
                      "onClick": undefined,
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "__emotion_base": "button",
                      "__emotion_forwardProp": undefined,
                      "__emotion_real": [Circular],
                      "__emotion_styles": Array [
                        Object {
                          "& svg": Object {
                            "height": "16px",
                            "width": "16px",
                          },
                          "height": "24px",
                          "width": "24px",
                        },
                      ],
                      "defaultProps": undefined,
                      "displayName": "Styled(button)",
                      "render": [Function],
                      "withComponent": [Function],
                    },
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p
                      className="filter-title"
                    >
                      Filters
                    </p>,
                    "className": "col-xs-6 pl8",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Filters",
                      "className": "filter-title",
                    },
                    "ref": null,
                    "rendered": "Filters",
                    "type": "p",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <button
                      className="reset-button fright"
                    >
                      RESET
                    </button>,
                    "className": "col-xs-4 pl0 pr0",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "RESET",
                      "className": "reset-button fright",
                      "onClick": undefined,
                    },
                    "ref": null,
                    "rendered": "RESET",
                    "type": "button",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "pt8 sbc5",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  0,
                  <div
                    className="fixed p8 sbcw b0 l0 r0 bs4"
                  >
                    <button
                      className="btn-filled-pri-large wfull"
                      type="button"
                    >
                      Apply Filter
                    </button>
                  </div>,
                ],
                "className": "filter-row-container block",
              },
              "ref": null,
              "rendered": Array [
                0,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <button
                      className="btn-filled-pri-large wfull"
                      type="button"
                    >
                      Apply Filter
                    </button>,
                    "className": "fixed p8 sbcw b0 l0 r0 bs4",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Apply Filter",
                      "className": "btn-filled-pri-large wfull",
                      "onClick": undefined,
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "Apply Filter",
                    "type": "button",
                  },
                  "type": "div",
                },
              ],
              "type": "form",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef(render)
            className="overflowh hide pbc1 radius100 fixed z14 r15 b15 flex alignCenter justifyCenter"
          >
            <IconSvg />
          </ForwardRef(render)>,
          <div
            className="col-xs-12 p0 z15 fixed t0 l0 overflowa pb64 b0 r0 sbc5 cityblockfilterContainer"
          >
            <div
              className="row row- sbcw filters-options-con"
            >
              <div
                className="row row- sbcw p15 flex alignCenter filters-header"
              >
                <div
                  className="col-xs-2 pl0"
                >
                  <ForwardRef(render)
                    className="close-filters flex alignCenter justifyCenter"
                    type="button"
                  >
                    <IconSvg />
                  </ForwardRef(render)>
                </div>
                <div
                  className="col-xs-6 pl8"
                >
                  <p
                    className="filter-title"
                  >
                    Filters
                  </p>
                </div>
                <div
                  className="col-xs-4 pl0 pr0"
                >
                  <button
                    className="reset-button fright"
                  >
                    RESET
                  </button>
                </div>
              </div>
              <div
                className="pt8 sbc5"
              />
              <form
                className="filter-row-container block"
              >
                0
                <div
                  className="fixed p8 sbcw b0 l0 r0 bs4"
                >
                  <button
                    className="btn-filled-pri-large wfull"
                    type="button"
                  >
                    Apply Filter
                  </button>
                </div>
              </form>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <IconSvg />,
            "className": "overflowh hide pbc1 radius100 fixed z14 r15 b15 flex alignCenter justifyCenter",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "button",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              Object {
                "& svg": Object {
                  "height": "25px",
                  "position": "absolute",
                  "width": "25px",
                },
                "boxShadow": "0 4px 12px #999",
                "height": "56px",
                "width": "56px",
              },
            ],
            "defaultProps": undefined,
            "displayName": "Styled(button)",
            "render": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row row- sbcw filters-options-con"
            >
              <div
                className="row row- sbcw p15 flex alignCenter filters-header"
              >
                <div
                  className="col-xs-2 pl0"
                >
                  <ForwardRef(render)
                    className="close-filters flex alignCenter justifyCenter"
                    type="button"
                  >
                    <IconSvg />
                  </ForwardRef(render)>
                </div>
                <div
                  className="col-xs-6 pl8"
                >
                  <p
                    className="filter-title"
                  >
                    Filters
                  </p>
                </div>
                <div
                  className="col-xs-4 pl0 pr0"
                >
                  <button
                    className="reset-button fright"
                  >
                    RESET
                  </button>
                </div>
              </div>
              <div
                className="pt8 sbc5"
              />
              <form
                className="filter-row-container block"
              >
                0
                <div
                  className="fixed p8 sbcw b0 l0 r0 bs4"
                >
                  <button
                    className="btn-filled-pri-large wfull"
                    type="button"
                  >
                    Apply Filter
                  </button>
                </div>
              </form>
            </div>,
            "className": "col-xs-12 p0 z15 fixed t0 l0 overflowa pb64 b0 r0 sbc5 cityblockfilterContainer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="row row- sbcw p15 flex alignCenter filters-header"
                >
                  <div
                    className="col-xs-2 pl0"
                  >
                    <ForwardRef(render)
                      className="close-filters flex alignCenter justifyCenter"
                      type="button"
                    >
                      <IconSvg />
                    </ForwardRef(render)>
                  </div>
                  <div
                    className="col-xs-6 pl8"
                  >
                    <p
                      className="filter-title"
                    >
                      Filters
                    </p>
                  </div>
                  <div
                    className="col-xs-4 pl0 pr0"
                  >
                    <button
                      className="reset-button fright"
                    >
                      RESET
                    </button>
                  </div>
                </div>,
                <div
                  className="pt8 sbc5"
                />,
                <form
                  className="filter-row-container block"
                >
                  0
                  <div
                    className="fixed p8 sbcw b0 l0 r0 bs4"
                  >
                    <button
                      className="btn-filled-pri-large wfull"
                      type="button"
                    >
                      Apply Filter
                    </button>
                  </div>
                </form>,
              ],
              "className": "row row- sbcw filters-options-con",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="col-xs-2 pl0"
                    >
                      <ForwardRef(render)
                        className="close-filters flex alignCenter justifyCenter"
                        type="button"
                      >
                        <IconSvg />
                      </ForwardRef(render)>
                    </div>,
                    <div
                      className="col-xs-6 pl8"
                    >
                      <p
                        className="filter-title"
                      >
                        Filters
                      </p>
                    </div>,
                    <div
                      className="col-xs-4 pl0 pr0"
                    >
                      <button
                        className="reset-button fright"
                      >
                        RESET
                      </button>
                    </div>,
                  ],
                  "className": "row row- sbcw p15 flex alignCenter filters-header",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <ForwardRef(render)
                        className="close-filters flex alignCenter justifyCenter"
                        type="button"
                      >
                        <IconSvg />
                      </ForwardRef(render)>,
                      "className": "col-xs-2 pl0",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": <IconSvg />,
                        "className": "close-filters flex alignCenter justifyCenter",
                        "onClick": undefined,
                        "type": "button",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "__emotion_base": "button",
                        "__emotion_forwardProp": undefined,
                        "__emotion_real": [Circular],
                        "__emotion_styles": Array [
                          Object {
                            "& svg": Object {
                              "height": "16px",
                              "width": "16px",
                            },
                            "height": "24px",
                            "width": "24px",
                          },
                        ],
                        "defaultProps": undefined,
                        "displayName": "Styled(button)",
                        "render": [Function],
                        "withComponent": [Function],
                      },
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <p
                        className="filter-title"
                      >
                        Filters
                      </p>,
                      "className": "col-xs-6 pl8",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Filters",
                        "className": "filter-title",
                      },
                      "ref": null,
                      "rendered": "Filters",
                      "type": "p",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <button
                        className="reset-button fright"
                      >
                        RESET
                      </button>,
                      "className": "col-xs-4 pl0 pr0",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "RESET",
                        "className": "reset-button fright",
                        "onClick": undefined,
                      },
                      "ref": null,
                      "rendered": "RESET",
                      "type": "button",
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "pt8 sbc5",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    0,
                    <div
                      className="fixed p8 sbcw b0 l0 r0 bs4"
                    >
                      <button
                        className="btn-filled-pri-large wfull"
                        type="button"
                      >
                        Apply Filter
                      </button>
                    </div>,
                  ],
                  "className": "filter-row-container block",
                },
                "ref": null,
                "rendered": Array [
                  0,
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <button
                        className="btn-filled-pri-large wfull"
                        type="button"
                      >
                        Apply Filter
                      </button>,
                      "className": "fixed p8 sbcw b0 l0 r0 bs4",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Apply Filter",
                        "className": "btn-filled-pri-large wfull",
                        "onClick": undefined,
                        "type": "button",
                      },
                      "ref": null,
                      "rendered": "Apply Filter",
                      "type": "button",
                    },
                    "type": "div",
                  },
                ],
                "type": "form",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
